### ISSUE TITLE
ValueError: could not broadcast input array from shape (0,) into shape (3,)

### ISSUE THREAD

**Issue by subodh-malgonde** [Gjacquenot/Puma-EM#1]
When the script `code/read_MLFMA_mesh_part1` is excuted the following error is thrown:

```
Traceback (most recent call last):
  File "code/read_MLFMA_mesh_part1.py", line 61, in <module>
    setup_mesh(params_simu, simuDirName)
  File "code/read_MLFMA_mesh_part1.py", line 33, in setup_mesh
    vertexes_coord, triangle_vertexes, triangles_physicalSurface = readMesh(path, name, params_simu)
  File "code/read_MLFMA_mesh_part1.py", line 9, in readMesh
    vertexes_coord, triangle_vertexes, triangles_physicalSurface = read_mesh_GMSH_2(os.path.join(path, name), params_simu.targetDimensions_scaling_factor, params_simu.z_offset)
  File "/Users/subodh/workspace/Puma-EM/code/read_mesh.py", line 201, in read_mesh_GMSH_2
    vertexes_coord[index, :] = list(map(float, tmp[1:]))
ValueError: could not broadcast input array from shape (0,) into shape (3,)
```

I am using python 3.8 and gmsh 4.8.4.

**Comment by subodh-malgonde** [comment]
Found the source of the error. The error on line number 201 here:

https://github.com/Gjacquenot/Puma-EM/blob/769b171adc89d44fa0c377fd332e273f26fdd68d/code/read_mesh.py#L197-L202

The `vertexes_coord` is an array of shape (1948, 3). So it needs 3 values per row. However in some cases only 1 values is present. Here is the explanation:

The file being opened is `simuDir/geo/cubi.msh.Nodes`. This files has lines such as:

```
0 1 0 1
1
0.1 0.1 -0.1
0 2 0 1
2
-0.1 0.1 -0.1
0 3 0 1
3
``` 
So some lines have 4 values whereas some lines have 1 value. For the first line variable `tmp` has 4 values and so `tmp[1:]` has 3 values. This works fine. However for the 2nd line `tmp` has 1 value and `tmp[1:]` is an empty list. This results in an error.

**Comment by subodh-malgonde** [comment]
Figured out the problem. The gmsh mesh format has seen some changes. The current format version is 4.1. Earlier versions were 3, 2.2, 2 and 1. 

The mesh being generated was of format 4.1, whereas this repo is written for formats 2.2 and older. I had to add the parameter `-format msh22` (i.e. generate a mesh of format 2.2) to the following line:

https://github.com/Gjacquenot/Puma-EM/blob/769b171adc89d44fa0c377fd332e273f26fdd68d/code/setup_GMSH.py#L34

Its working now.
